pulp_ansible/app/global_access_conditions.py

```diff
@@ -9,7 +9,13 @@ def v3_can_view_repo_content(request, view, action):
 
     if "distro_base_path" in view.kwargs:
         distro_base_path = view.kwargs["distro_base_path"]
-        repo = models.AnsibleDistribution.objects.get(base_path=distro_base_path).repository.cast()
+        distro = models.AnsibleDistribution.objects.select_related(
+            "repository", "repository_version"
+        ).get(base_path=distro_base_path)
+        if not (repo := distro.repository.cast()):
+            if not (repo_ver := distro.repository_version.cast()):
+                return False
+            repo = repo_ver.repository
```

Comment:
```suggestion
        if not (repo := distro.repository):
            if not (repo_ver := distro.repository_version):
                return False
            repo = repo_ver.repository
        repo = repo.cast()
```
You can't call `cast` on a repository version and you can't call it on a `None` object.

Reply:
Doh.